cromwell_tools/_cromwell_api.py

```diff
@@ -0,0 +1,425 @@
+        # TODO: Add `raise_for_status` flag to this method and remove the hard-coded `raise_for_status()` call
+        # TODO: Purifying the methods by taking the specific retry policy out from this method
+
+        Args:
+            wdl_file (_io.BytesIO or str): The workflow source file to submit for execution. From version 35,
+                Cromwell starts to accept URL to the WDL file besides actual WDL files.
+            inputs_json (_io.BytesIO): File-like object containing input data in JSON format.
+            options_json (Optional[_io.BytesIO]): Cromwell configs file.
+            inputs2_json (Optional[_io.BytesIO]): Inputs file 2.
+            dependencies_json (Optional[_io.BytesIO]): Zip file containing dependencies.
+            auth (cromwell_tools._cromwell_auth.CromwellAuth): authentication class holding headers or
+                auth information to a Cromwell server.
+            collection_name (Optional[str]): Collection in SAM that the workflow should belong to.
+            label (Optional[Union[str, _io.BytesIO]]): JSON file containing a collection of
+                key/value pairs for workflow labels. # TODO: verify these types are accurate
+            validate_labels (Optional[bool]) If True, validate cromwell labels. (default False)
+
+        Raises:
+            requests.exceptions.HTTPError: This will be raised when raise_for_status is True and Cromwell returns
+                a response that satisfies 400 <= response.status_code < 600.
+
+        Returns:
+            requests.Response: HTTP response from Cromwell.
+        """
+        manifest = prepare_workflow_manifest(
+                wdl_file, inputs_json, dependencies_json, options_json, inputs2_json)
+
+        if validate_labels and label is not None:
+            validate_cromwell_label(label)
+
+        files = {
+            'workflowSource': wdl_file,
+            'workflowInputs': inputs_json,
+        }
+
+        # TODO: these three checks should be done inside prepare workflow manifest?
+        if 'inputs2_json' in manifest:
+            files['workflowInputs_2'] = manifest['inputs2_json']
+        if 'dependencies_json' in manifest:
+            files['workflowDependencies'] = manifest['dependencies_json']
+        if 'options_json' in manifest:
+            files['workflowOptions'] = manifest['options_json']
+
+        if label:
+            files['labels'] = label
+        if collection_name:  # TODO: this check should happen elsewhere, related to caas
+            files['collectionName'] = collection_name
+
+        response = cls.requests.post(url=auth.url + cls._workflow_endpoint,
+                                     files=files,
+                                     auth=auth.auth,
+                                     headers=auth.header)
+
+        response.raise_for_status()
+        return response
+
+    @classmethod
+    def wait(cls, workflow_ids, timeout_minutes, auth, poll_interval_seconds=30, verbose=True):
```

Comment:
We can set the default `timeout_minutes` to 120 here so that it is consistent with the behavior of the CLI

Reply:
👍 Will do!